core/user-flow.js

```diff
@@ -10,9 +10,46 @@ import {startTimespanGather} from './gather/timespan-runner.js';
 import {navigationGather} from './gather/navigation-runner.js';
 import {Runner} from './runner.js';
 import {initializeConfig} from './config/config.js';
+import {getFormatted} from '../shared/localization/format.js';
+import * as i18n from './lib/i18n/i18n.js';
 
 /** @typedef {WeakMap<LH.UserFlow.GatherStep, LH.Gatherer.FRGatherResult['runnerOptions']>} GatherStepRunnerOptions */
 
+const UIStrings = {
+  /**
+   * @description Default name for a user flow on the given url. "User flow" refers to the series of page navigations and user interactions being tested on the page.
+   * @example {https://example.com} url
+   */
+  defaultFlowName: 'User flow ({url})',
+  /**
+   * @description Default name for a user flow step that analyzes a page navigation.
+   * @example {https://example.com} url
+   */
+  defaultNavigationName: 'Navigation report ({url})',
+  /**
+   * @description Default name for a user flow step that analyzes user interactions over a period of time.
+   * @example {https://example.com} url
+   */
+  defaultTimespanName: 'Timespan report ({url})',
+  /**
+   * @description Default name for a user flow step that analyzes the page state at a point in time.
+   * @example {https://example.com} url
+   */
+  defaultSnapshotName: 'Snapshot report ({url})',
+};
+
+const str_ = i18n.createIcuMessageFn(import.meta.url, UIStrings);
+
+/**
+ * @param {string} message
+ * @param {Record<string, string | number>} values
+ * @param {LH.Locale} locale
+ */
+function translate(message, values, locale) {
+  const icuMessage = str_(message, values);
+  return getFormatted(icuMessage, locale);
```

Comment:
Feels weird to be immediately removing the IcuMessage. Should this be handled by either the flow report's i18n or by the LHR's typical `replaceIcuMessages` flow?

This won't work with swapping locales, but looks like there's nowhere you can swap a flow report's locale anyways.

Reply:
The messages are designed to be customized, so I want to avoid shipping ICU messages as the step/flow names.